### PR TITLE
[6.2.z][cherry-pick] qe test coverage 1398392

### PR DESCRIPTION
**Pull request by ldjebran**
BZ : https://bugzilla.redhat.com/show_bug.cgi?id=1398392
original PR: https://github.com/SatelliteQE/robottelo/pull/4634
resolution show message "Invalid smart-proxy id"
nightly show message "Invalid capsule id"
resolved in https://github.com/shlomizadok/foreman/blob/aaeeeb33b03f263d36b71cdab7a06170bc90462b/app/controllers/api/v2/domains_controller.rb#L64

test decorated as affected on 6.2.z
test console result
```console
2017-05-11 10:41:25 - robottelo.ssh - INFO - >>> LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv domain create --name="qCWLVGdjYt" --dns-id="-1"
2017-05-11 10:41:27 - robottelo.ssh - INFO - <<< stderr
[ERROR 2017-05-11 03:41:27 Exception] ERROR:  insert or update on table "domains" violates foreign key constraint "domains_dns_id_fk"
DETAIL:  Key (dns_id)=(-1) is not present in table "smart_proxies".

Could not create the domain:
  ERROR:  insert or update on table "domains" violates foreign key constraint "domains_dns_id_fk"
  DETAIL:  Key (dns_id)=(-1) is not present in table "smart_proxies".
[ERROR 2017-05-11 03:41:27 Exception] 

```

### CODE REVIEW

File: tests/foreman/cli/test_domain.py [hunk 1]
```diff
@@ -184,6 +190,34 @@ def test_negative_create(self):
...
+                'dns-id': -1,
+            })
+        valid_messages = ['Invalid smart-proxy id', 'Invalid capsule id']
+        exception_string = str(context.exception)
+        messages = [
+            message
+            for message in valid_messages
+            if message in exception_string
+        ]
+        self.assertGreater(len(messages), 0)
```
Comment (oshtaier) [inline tests/foreman/cli/test_domain.py]: isn't it a static number of messages? can't we assume `self.assertEqual(len(messages), 2)`?

Comment (oshtaier) [inline tests/foreman/cli/test_domain.py]: Ohh, now I see
```
resolution show message "Invalid smart-proxy id"
nightly show message "Invalid capsule id"
```
So, in nightly there is one message and in 6.2.x is another. Then we need to submit in master and 6.2.z branches code with different messages

Comment (ldjebran) [inline tests/foreman/cli/test_domain.py]: @oshtaier in reality I do not know which message should be, as in nightly the test pass with message  "Invalid capsule id", but in the resolution linked with the bug the message is  "Invalid smart-proxy id" so the messages differ in some versions and one of them can affect any build, as the two messages are valid, I decided to include them both.

Comment (ldjebran) [inline tests/foreman/cli/test_domain.py]: @oshtaier investigated a bit on the diff messages and came to the logic that "Invalid smart-proxy id" get's translated some how to "Invalid capsule id" , unfortunately was not able to find the po file

Comment (oshtaier) [inline tests/foreman/cli/test_domain.py]: Okay, so still not sure what we should do.
If `"Invalid capsule id"` is specific for nightly, so submit PR with that message into nightly branch and PR with `"Invalid smart-proxy id"` into master or whatever order is needed, but fact is that we can have only one message per one build

**Comment by ldjebran** [comment]
@oshtaier code updated to include only one message

**Comment by oshtaier** [comment]
Hmm, a little bit confusing to get message about smart proxy id when providing wrong dns id, but probably okay

ACK

**Pull request merged**